.storybook/lucid-docs-addon/util.js

```diff
@@ -14,3 +16,152 @@ export const stripIndent = text => {
 		.map(line => line.replace(minimumLeadingWhitespace, ''))
 		.join('\n');
 };
+
+export const formatSource = source =>
+	source
+		.replace(/(\.\.\/)*(src\/)?index(\.js)?/, 'lucid-ui')
+		.replace(/\t/g, '  ')
+		.replace(/\/\/\s?begin-hide-from-docs[\s\S]*\/\/\s?end-hide-from-docs/, '')
+		.replace(/\n*$/, '\n');
+
+// custom props parsing for storybook DocsPage
+// our usage of prop types and peek break some assumptions that storybook makes about how props info is passed
+// this function tries to get prop type info from typescript (via react-docgen-typescript-loader) for TS components
+// then falls back on propTypes for non-TS components
+// in addition, it handles setting default values based on defaultProps
+export const getPropDefs = component =>
+	_.has(component, '__docgenInfo')
+		? getPropsFromDocgen(component)
+		: getPropsFromPropTypes(component);
+
+const getPropsFromDocgen = component =>
+	_.map(
+		component.__docgenInfo.props,
+		({ type, required, description, defaultValue }, property) =>
+			mergePropwithDefaults(component, {
+				type,
+				required,
+				description,
+				defaultValue,
+				name: property,
+			})
+	);
+
+const getPropsFromPropTypes = component =>
+	_.map(
+		getPropsData(component),
+		({ name, type, isRequired, dynamicData, text, defaultValue }) =>
+			mergePropwithDefaults(component, {
+				name,
+				type: {
+					name: propTypeName({ type, dynamicData }),
+				},
+				required: isRequired,
+				description: text,
+				defaultValue,
+			})
+	);
+
+const mergePropwithDefaults = (
+	component,
+	{ name, type, required, description, defaultValue }
+) => {
+	// compare with defaultProps to set props optional and set default value as needed
+	const defaultValueValue =
+		_.get(defaultValue, 'defaultValue.value') ||
+		getDefaultPropValue(component, name);
+	return {
+		name,
+		type,
+		required: required && _.isNil(defaultValueValue),
+		description,
+		defaultValue: defaultValueValue,
+	};
+};
+
```

Comment:
The PropTypes parsing functions from lucid-docs-addon were moved into this utils file so that the storybook docs addon could use them as well (for components not yet converted to TS).